.github/workflows/uploadFirrtlBinaries.yml

```diff
@@ -42,6 +42,26 @@ jobs:
         with:
           fetch-depth: 2
           submodules: "true"
+      
+      # Package up sources for distribution, as the default source bundles from GitHub don't include LLVM. Only runs on the ubuntu-20.04 runner.
+      - name: Archive Sources
+        if: ${{ matrix.runner }} == ubuntu-20.04 && startsWith(github.ref, 'refs/tags/')
+        run: |
+          touch circt-full-sources.tar.gz
+          tar \
+            --exclude-vcs \
+            --exclude=circt-full-sources.tar.gz \
+            -czf \
+            circt-full-sources.tar.gz .
+      - name: Upload Source Archive
+        uses: AButler/upload-release-assets@v2.0
+        if: ${{ matrix.runner }} == ubuntu-20.04 && startsWith(github.ref, 'refs/tags/')
+        with:
+          files: circt-full-sources.tar.gz
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Delete Sources Archive
+        if: ${{ matrix.runner }} == ubuntu-20.04 && startsWith(github.ref, 'refs/tags/')
+        run: rm circt-full-sources.tar.gz
```

Comment:
Should this be a separate workflow or a separate job within the workflow? This actual task seems to have little to do with the rest of what this workflow does so it might be worth separating the concerns.

Reply:
I like that this uses the actual sources, and I see this workflow as "Upload Firth Release Artifacts". Maybe I should rename it?